packages/types/src/index.ts

```diff
@@ -87,63 +81,81 @@ export interface MrkdwnElement {
 
 export interface MrkdwnOption {
   text: MrkdwnElement;
-  value?: string;
+  value: string;
   url?: string;
   description?: PlainTextElement;
 }
 
 export interface PlainTextOption {
   text: PlainTextElement;
-  value?: string;
+  value: string;
   url?: string;
   description?: PlainTextElement;
 }
 
-export type Option = MrkdwnOption | PlainTextOption;
-
 export interface Confirm {
-  title?: PlainTextElement;
+  title: PlainTextElement;
   text: PlainTextElement | MrkdwnElement;
-  confirm?: PlainTextElement;
-  deny?: PlainTextElement;
-  style?: 'primary' | 'danger';
+  confirm: PlainTextElement;
+  deny: PlainTextElement;
+  style?: 'primary' | 'danger'; // TODO: factor out into own type
+}
+
+export type Option = MrkdwnOption | PlainTextOption;
+
+export interface DispatchActionConfig {
+  trigger_actions_on?: ('on_enter_pressed' | 'on_character_entered')[];
 }
 
 /*
- * Action Types
+ * Block Elements: https://api.slack.com/reference/block-kit/block-elements
+ * Also known as interactive elements
+ * Exported below as Action
  */
 
+export type KnownAction =
+  Select | MultiSelect | Button | Overflow | Datepicker | Timepicker | RadioButtons | Checkboxes | PlainTextInput;
+
+export interface Action {
+  type: string;
+  /**
+   * @description: A string acting as a unique identifier for a block.
+   */
+  block_id?: string; // TODO: in event payloads, this property will always be present. When defining a block, however,
+  // it is optional. If not defined when the block is created, Slack will auto-assign a block_id for you - thus why it
+  // is always present in payloads. So: how can we capture that? Differentiate between the event payload vs. the block
+  // element?
+}
+
 // Selects and Multiselects are available in different surface areas so I've separated them here
 export type Select = UsersSelect | StaticSelect | ConversationsSelect | ChannelsSelect | ExternalSelect;
 
 export type MultiSelect =
   MultiUsersSelect | MultiStaticSelect | MultiConversationsSelect | MultiChannelsSelect | MultiExternalSelect;
 
-export interface Action {
-  type: string;
-  action_id?: string;
-}
-
 export interface UsersSelect extends Action {
   type: 'users_select';
+  action_id: string;
   initial_user?: string;
-  placeholder?: PlainTextElement;
+  placeholder: PlainTextElement;
   confirm?: Confirm;
   focus_on_load?: boolean;
 }
 
 export interface MultiUsersSelect extends Action {
   type: 'multi_users_select';
+  action_id: string;
   initial_users?: string[];
-  placeholder?: PlainTextElement;
+  placeholder: PlainTextElement;
   max_selected_items?: number;
   confirm?: Confirm;
   focus_on_load?: boolean;
 }
 
 export interface StaticSelect extends Action {
   type: 'static_select';
-  placeholder?: PlainTextElement;
+  placeholder: PlainTextElement;
+  action_id: string;
```

Comment:
nit: when you build a block element, setting action_id is optional (auto-generated one will be used in the case). thus, strictly speaking, this can be optional

Reply:
Are you sure `action_id` is optional? The docs say it is required. I know `block_id` is optional and one will be auto-generated for you in that case.
I will test this today in real apps and report back!

Reply:
Wow I can't believe it, action_id _is_ optional 😆